packages/harvester/test/MockHarvester.test.ts

```diff
@@ -72,4 +87,62 @@ describe("MockHarvester", () => {
 
     expect(gasUsed).to.be.lte(gasLimit);
   });
+
+  it("should not harvest because of canHarvestVault function", async () => {
+    await harvester.setCanHarvestVault(false);
+    await harvester.setShouldHarvestVault(true);
+    const { upkeepNeeded_ } = await harvester.checkUpkeep([]);
+    await harvester.setCanHarvestVault(true);
+    await harvester.setShouldHarvestVault(true);
+    await harvester.setHarvestVault(true);
+    expect(upkeepNeeded_).to.be.false;
+  });
+
+  it("should not harvest because of shouldHarvestVault function", async () => {
+    await harvester.setCanHarvestVault(true);
+    await harvester.setShouldHarvestVault(false);
+    const { upkeepNeeded_ } = await harvester.checkUpkeep([]);
+    await harvester.setCanHarvestVault(true);
+    await harvester.setShouldHarvestVault(true);
+    await harvester.setHarvestVault(true);
+    expect(upkeepNeeded_).to.be.false;
+  });
+
+  it("should not harvest because of canHarvestVault and shouldHarvestVault", async () => {
```

Comment:
What are we testing here? Seems unnecessary to test boolean logic.

Reply:
That's true. Theoretically boolean AND can't fail but in the future somebody can change the code in the contract. I am testing behavior not that the code in just that moment can't fail if the previous two tests pass. The tests are to use them and test and future changes. 99% from the bugs are because somebody say "that can't fail, there is no need to test it" and after somebody change the code and there is no test for it and here you have the problem.
If you think it's unnecessary for our needs I can remove that test?

Reply:
This is practically the same test as the previous two. If anything is changed in `_willHarvest` they will fail too. 

Not sure what other future scenarios you have in mind.